### PR TITLE
Move WandB loggers into spacy-loggers

### PR DESCRIPTION
**Pull request by rspeer**
This PR depends on https://github.com/explosion/spacy-loggers being published.

## Description
Move the Weights & Biases logger into `spacy-loggers`, a new dependency of spaCy.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

This won't pass CI tests until there's a version of spacy-loggers available.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Our libraries
 spacy-legacy>=3.0.7,<3.1.0
+spacy-loggers>=1.0,<2.0
```
Comment (adrianeboyd) [inline requirements.txt:3]: Just for consistency:

```suggestion
spacy-loggers>=1.0.0,<2.0.0
```

File: setup.cfg [hunk 2]
```diff
@@ -41,6 +41,7 @@ setup_requires =
 install_requires =
     # Our libraries
     spacy-legacy>=3.0.7,<3.1.0
+    spacy-loggers>=1.0,<2.0
```
Comment (adrianeboyd) [inline setup.cfg:44]:
```suggestion
    spacy-loggers>=1.0.0,<2.0.0
```

**Review by adrianeboyd — CHANGES_REQUESTED** [review]

File: website/docs/api/top-level.md [hunk 3]
```diff
@@ -463,62 +463,6 @@ start decreasing across epochs.
 
  </Accordion>
 
-#### spacy.WandbLogger.v2 {#WandbLogger tag="registered function"}
```
Comment (svlandeg) [inline website/docs/api/top-level.md:466]: I'm guessing that this anchor point `#WandbLogger` is being referred to from other places of the docs too, and we'll need to update those too. Perhaps we can make sure that https://github.com/explosion/spacy-loggers#wandb or something such is a valid URL point to the latest `WandbLogger` implementation, then update the links through the docs?

We'll also need to update comments like "There are two built-in loggers" --> one built-in, with alternatives defined in https://github.com/explosion/spacy-loggers, etc.

**Comment by adrianeboyd** [comment]
This will need to updated following #9296 to remove the v3 logger cleanly, too.

**Comment by svlandeg** [comment]
I'll just put this in draft for now - @rspeer feel free to take it back out of draft when you're ready!

**Review by svlandeg — APPROVED** [review]
Looks good to me! :-)

**Review by adrianeboyd — APPROVED** [review]

**Pull request merged**